src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java

```diff
@@ -2256,7 +2256,11 @@ public void connectionSendsPingsRegularly() throws InterruptedException {
 
         hubConnection.start().timeout(1, TimeUnit.SECONDS).blockingAwait();
 
-        TimeUnit.MILLISECONDS.sleep(100);
+        String message = mockTransport.getNextSentMessage().timeout(1, TimeUnit.SECONDS).blockingGet();
+        assertEquals("{\"type\":6}" + RECORD_SEPARATOR, message);
+        message = mockTransport.getNextSentMessage().timeout(1, TimeUnit.SECONDS).blockingGet();
+        assertEquals("{\"type\":6}" + RECORD_SEPARATOR, message);
+
         hubConnection.stop().timeout(1, TimeUnit.SECONDS).blockingAwait();
 
         String[] sentMessages = mockTransport.getSentMessages();
```

Comment:
Nit: isn't the rest of the test now redundant since you already know at least two pings were sent?

Reply:
Yeah that's true 😆